### PR TITLE
Feat/control input update

### PR DESCRIPTION
**Pull request by aniketmpatil**
- added control inputs to the model
- added subscriber for teleop values
- changed state transition matrix to incorporate control matrix

### CODE REVIEW

File: src/modules/localization/interface/src/localization_publisher.hpp [hunk 1]
```diff
@@ -53,6 +55,8 @@ class EKFPublisher : public rclcpp::Node
     void imu_callback(const sensor_msgs::msg::Imu::SharedPtr msg);
 
     void odom_callback(const nav_msgs::msg::Odometry::SharedPtr msg);
+
+    void control_callback(const geometry_msgs::msg::Twist::SharedPtr msg);
```
Comment (pinakjani) [inline src/modules/localization/interface/src/localization_publisher.hpp]: Add doxygen comments.

File: src/modules/localization/domain/dynamic_model/constant_acceleration_model.hpp [hunk 2]
```diff
@@ -51,6 +51,7 @@ namespace localization::dynamic_model {
          */
         void update_state_matrix(Eigen::MatrixXd& F, double dt);
 
+
```
Comment (pinakjani) [inline src/modules/localization/domain/dynamic_model/constant_acceleration_model.hpp:54]: Remove unnecessary additions

**Review by pinakjani — CHANGES_REQUESTED** [review]
Requested minor changes

File: src/modules/localization/interface/src/localization_publisher.hpp (hunk 1 above)
Comment (aniketmpatil) [reply src/modules/localization/interface/src/localization_publisher.hpp]: Added for all 3 callback functions

File: src/modules/localization/domain/dynamic_model/constant_acceleration_model.hpp (hunk 2 above)
Comment (aniketmpatil) [reply src/modules/localization/domain/dynamic_model/constant_acceleration_model.hpp:54]: Removed

**Review by pinakjani — APPROVED** [review]
Approving for merge

**Pull request merged**